pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -450,41 +477,68 @@ func (i *Instances) AddSSHKeyToAllInstances(user string, keyData []byte) error {
 	return errors.New("unimplemented")
 }
 
-func (i *Instances) NodeAddresses(name string) ([]api.NodeAddress, error) {
-	glog.V(4).Infof("NodeAddresses(%v) called", name)
-
-	addrs, err := getAddressesByName(i.compute, name)
+// This is only ever called locally in kubelet
+func (i *Instances) NodeAddresses(_ string) ([]api.NodeAddress, error) {
+	glog.V(4).Infof("NodeAddresses() called")
+	addrs, err := getAddressesByName(i.os.compute, i.os.openStackInstance.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	glog.V(4).Infof("NodeAddresses(%v) => %v", name, addrs)
+	glog.V(4).Infof("NodeAddresses(%v) => %v", addrs)
 	return addrs, nil
 }
 
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
 func (i *Instances) ExternalID(name string) (string, error) {
-	srv, err := getServerByName(i.compute, name)
+	if i.os.openStackInstance.Hostname == name {
```

Comment:
So the previous code (unless it's been broken recently) was careful to *not* make the assumption that k8s node name (the `name` argument here) was the same as the hostname.  Specifically, the previous code assumed openstack instance name was unique (at least amongst ACTIVE VM instances) and == the k8s node name (see `getServerByName`).

There were numerous issues previously with the k8s/cinder plugin caused by more and more workarounds to try to deal with the initial (incorrect!) assumption that instance UUID could be eventually derived from local hostname, which just isn't true on openstack.  I'd like it if we didn't restore any part of that logic without a super-strong reason...